python/lzreposync/src/lzreposync/__init__.py

```diff
@@ -78,18 +80,55 @@ def main():
         type=str,
     )
 
+    parser.add_argument(
+        "--channel",
+        help="The channel id of which you want to synchronize repositories",
+        dest="channel",
+        type=int,
+        default=None,
+    )
+
+    # TODO encapsulate everything in a class LzRepoSync
+
     args = parser.parse_args()
     arch = args.arch
     if arch != ".*":
         # pylint: disable-next=consider-using-f-string
         arch = "(noarch|{})".format(args.arch)
 
     logging.getLogger().setLevel(args.loglevel)
-    rpm_repository = RPMRepo(
-        args.name, args.cache, args.url, arch
-    )  # TODO args.url should be args.repo, no ?
-    packages = rpm_repository.get_packages_metadata()  # packages is a generator
-    failed = 0
-    for i, batch in enumerate(batched(packages, args.batch_size)):
-        failed += import_package_batch(batch, i)
-    logging.debug("Completed import with %d failed packages", failed)
+    if args.url:
+        rpm_repository = RPMRepo(args.name, args.cache, args.url, arch)
+        packages = rpm_repository.get_packages_metadata()  # packages is a generator
+        failed = 0
+        for i, batch in enumerate(batched(packages, args.batch_size)):
+            failed += import_package_batch(batch, i)
+        logging.debug("Completed import with %d failed packages", failed)
+
+    else:
+        # No url specified
+        if args.channel:
+            channel_id = args.channel
+            target_repos = db_utils.get_repositories_by_channel_id(channel_id)
+            for repo in target_repos:
+                if repo.repo_type == "yum":
+                    rpm_repository = RPMRepo(
+                        repo.repo_label, args.cache, repo.source_url, arch
```

Comment:
Use the channel arch instead of `arch` here: the architecture for a channel is provided by the database.